stix_shifter_modules/paloalto/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,477 @@
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, ComparisonComparators, Pattern, CombinedComparisonExpression, CombinedObservationExpression
+import logging
+import re
+from datetime import datetime, timedelta
+from os import path
+import json
+
+logger = logging.getLogger(__name__)
+
+START_STOP_PATTERN = r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z)"
+MAC = '^(([0-9a-fA-F]{2}[:-]){5}([0-9a-fA-F]{2}))$'
+CONFIG_MAP_PATH = "json/config_map.json"
+FIELDS_MAP_PATH = "json/fields_map.json"
+
+STOP_TIME = datetime.utcnow()
+
+
+class StartStopQualifierValueException(Exception):
+    pass
+
+
+class QueryStringPatternTranslator:
+    """
+    translate stix pattern to native data source query language
+    """
+    def __init__(self, pattern: Pattern, data_model_mapper, options):
+        logger.info("Palo Alto Cortex XDR Connector")
+        self.dmm = data_model_mapper
+        self.options = options
+        self.timeframe = []
+        self.comparator_lookup = self.dmm.map_comparator()
+        self.config_map = self.load_json(CONFIG_MAP_PATH)
+        self.all_fields_map = self.load_json(FIELDS_MAP_PATH)
+        self.translated_query = self.parse_expression(pattern)
+        self.qualified_query = self._create_formatted_query(self.dmm, self.translated_query,
+                                                            self.timeframe, self.all_fields_map, options)
+
+    @staticmethod
+    def load_json(rel_path_of_file):
+        """
+        Consumes a json file and returns a dictionary
+        :param rel_path_of_file: str
+        :return: dict
+        """
+        _json_path = path.dirname(path.realpath(__file__)) + "/" + rel_path_of_file
+        if path.exists(_json_path):
+            with open(_json_path, encoding='utf-8') as f_obj:
+                return json.load(f_obj)
+        raise FileNotFoundError
```

Comment:
you should use try/except block and raise proper `FileNotFoundError` exception

Reply:
Code changes has been done to handle with try and except.